File: First Missing Positive/First_Missing_Positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int n = nums.length;
        for (int i = 1; i <= n; i++) {
            if (!set.contains(i)) return i;
        }
        return n + 1;
    }
}

File: First Missing Positive/First_Missing_Positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_set<int> st;
        for(auto i: nums) {
            st.insert(i);
        }
        int n = nums.size();
        for(int i =1; i<=n; i++) {
            if(st.count(i) == 0) return i;
        }
        return n+1;
    }
};